### ISSUE TITLE
Hiccups when closed channels are being forgotten

### ISSUE THREAD

**Issue by michael1011** [ElementsProject/lightning#7962]
When closed channels with lots of updates are being forgotten, the CLN RPC freezes until the deletion is done. 

From @rustyrussell:

> I strongly suspect it's this:
>	stmt = db_prepare_v2(w->db, SQL("DELETE FROM channel_htlcs "
					"WHERE channel_id=?"));

> This is because there's no index for that exact thing, so this causes a sweep of the entire table.  Fix is obvious: add another index...

**Comment by whitslack** [comment]
Do you not have a unique index on `(channel_id, channel_htlc_id, direction)`? My `channel_htlcs` table has that index.

**Comment by michael1011** [comment]
Yes, that index exists. And even it didn't, I wouldn't want to start creating indexes myself by hand

**Comment by whitslack** [comment]
> Yes, that index exists.

So there would be nothing to gain by adding an index on solely `channel_id`. It would only slow down writes.

**Comment by rustyrussell** [comment]
This is a good point: I suspect the issue is elsewhere, so I'll need to actually reproduce.
